nailgun/entities.py

```diff
@@ -2334,6 +2355,23 @@ def read(self, entity=None, attrs=None, ignore=None):
         attrs['path_'] = attrs.pop('path')
         return super(Media, self).read(entity, attrs, ignore)
 
+    def update(self, fields=None):
+        """Fetch a complete set of attributes for this entity.
+
+        Beware of `Bugzilla #1236008
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1236008>`_:
+        "Cannot use HTTP PUT to associate location with media"
+
+        """
+        self.update_json(fields)
+        return self.read()
```

Comment:
Thanks for clarifying on IRC.. that's a different issue. I'll file a new bz and will update the PR